03-Bootstrap/assignment/grid.html

```diff
@@ -0,0 +1,213 @@
+          <h1 class="display-4 text-center">Assignment 03 - Bootstrap 4</h1>
+        </header>
+
+        <nav
+          class="navbar navbar-expand-lg navbar-light bg-info d-flex justify-content-center py-3"
+        >
+          <ul class="navbar-nav">
+            <li class="nav-item active">
+              <a class="nav-link" href="index.html">Home</a>
+            </li>
+            <li class="nav-item">
+              <a class="nav-link" href="images.html">Images</a>
+            </li>
+            <li class="nav-item">
+              <a class="nav-link" href="accordion.html">Accordion</a>
+            </li>
+            <li class="nav-item">
+              <a class="nav-link" href="grid.html">Grid</a>
+            </li>
+          </ul>
+        </nav>
+      </div>
+
+      <div class="container">
+        <div class="row mt-3 mb-3">
+          <div class="col-xs-12 col-sm-12 col-md-4 col-lg-3 col-xl-2">
+            <div class="card">
+              <div class="card-body">
+                <h5 class="card-title">Card 1</h5>
+                <p class="card-text">
+                  This is a wider card with supporting text below as a natural
+                  lead-in to additional content. This content is a little bit
+                  longer.
+                </p>
+              </div>
+            </div>
+          </div>
+
+          <div class="col-xs-12 col-sm-12 col-md-4 col-lg-3 col-xl-2">
+            <div class="card">
+              <div class="card-body">
+                <h5 class="card-title">Card 2</h5>
+                <p class="card-text">
+                  This is a wider card with supporting text below as a natural
+                  lead-in to additional content. This content is a little bit
+                  longer.
+                </p>
+              </div>
+            </div>
+          </div>
+
+          <div class="col-xs-12 col-sm-12 col-md-4 col-lg-3 col-xl-2">
+            <div class="card">
+              <div class="card-body">
+                <h5 class="card-title">Card 3</h5>
+                <p class="card-text">
+                  This is a wider card with supporting text below as a natural
+                  lead-in to additional content. This content is a little bit
+                  longer.
+                </p>
+              </div>
+            </div>
+          </div>
+
+          <div class="col-xs-12 col-sm-12 col-md-4 col-lg-3 col-xl-2">
+            <div class="card">
+              <div class="card-body">
+                <h5 class="card-title">Card 4</h5>
+                <p class="card-text">
+                  This is a wider card with supporting text below as a natural
+                  lead-in to additional content. This content is a little bit
+                  longer.
+                </p>
+              </div>
+            </div>
+          </div>
+
+          <div class="col-xs-12 col-sm-12 col-md-4 col-lg-3 col-xl-2">
```

Comment:
One refinement you can do on these is get rid of the double 12 classes and just use one. It's not wrong to be pedantic like this, but since you're using `col-sm-12` than anything smaller than that should use 12 cols too. Just a tip for the future to try and use less code if you can.